ui: Node list view redesign - Fix follow up issues

koorosh opened the pull request:
Current PR contains fixes on following issues: https://github.com/cockroachdb/cockroach/issues/43883

- Dropdown component refactored to include default button as a toggle component and
added render prop to provide own implementation of toggle component
- Button components are extended with ability to render Icon within the button
- Fixed stylings on Cluster overview page

koorosh committed:
ui: Rename table name from 'Live Nodes' to 'Nodes'

Release note (admin ui change): Cluster Overview view changed

koorosh committed:
ui: Rename status for aggregated rows in Nodes table on Cluster Overview page

- Status name is changed from 'Ready' to 'Live'

Release note (admin ui change): Cluster Overview view changed

koorosh committed:
ui: Fix Node List/Map toggle view

- Refactored Dropdown component to include default button implementation.
Added ability to provide own implementation with render prop (dropdownToggleButton)
- Added icon rendering for Button component

Release note (admin ui change): Cluster Overview view changed

koorosh committed:
ui: Fix alignments and spacing for Cluster Overview page

- Rename nodes to N{nodeId} format according to provided styles;
- Highlight links in Nodes table when hover on entire row;
- Align section titles by center;

Release note (admin ui change): Cluster Overview view changed

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/44212)
<!-- Reviewable:end -->

koorosh committed:
ui: Fix linting error

Release note: None

dhartunian commented:
@Koorosh sorry I commented on issue and not the PR. Can you rebase this and make the small change to the triangle icon on menu as clarified by Anne in the issue?

koorosh committed:
ui: Always show dropdown button with caret down icon

Toggle icon should always be displayed as cared down.

Release note: None

Koorosh commented:
> @Koorosh sorry I commented on issue and not the PR. Can you rebase this and make the small change to the triangle icon on menu as clarified by Anne in the issue?

Done.

Annebirzin commented:
@Koorosh This is looking great! I noticed a few tiny adjustments that could be made (these are not critical and can be a separate issue if needed)

**Node statuses:**
- [ ] Make all node status fonts bold

**Node names:**
- [ ] Make all node numbers bold - just the number not the entire name (ie. **N1** us-east1)

**Node list module:**
- [ ] Add space below the last node and the white bound box (so there is some white space below the list) 
![Screen Shot 2020-02-05 at 4 04 07 PM](https://user-images.githubusercontent.com/51920721/73884011-fbe6e800-4832-11ea-82d4-1599a9b5cd86.png)

**Node list/map selection bar**
- [ ] The height of the bar seems to change when I switch from list to map. Wondering if we can keep these at the same height (using the node list height)

**Node/decommissioned node table headers:**
- [ ] the font still seems to be ‘regular’ not ‘semi-bold’ (ideally we can match the statement page table header)
- [ ] node table header: the background grey should be #F5F7FA 
- [ ] update ‘# of nodes’ column header to ‘node count’ 
![Screen Shot 2020-02-05 at 3 21 52 PM](https://user-images.githubusercontent.com/51920721/73884060-13be6c00-4833-11ea-926b-9fd8288150ef.png)

koorosh committed:
ui: Adjust styles for Cluster Overview page

- Adjust header font sizes for nodes tables;
- Make table columns as fixed to properly allocate space for
main columns and prevent resizing after rows expanded.
- Add tests on Live Nodes view which were mostly affected by current
change
- Add test-util/connectedMount helper function, it wraps component under
test with redux-store and router components to be able test components
which depend on Store and routers navigation.

Release note: None

koorosh commented:
@Annebirzin , @danhhz 
Done, adjustments are included in this PR.

Annebirzin commented:
@koorosh I think you meant to tag @dhartunian ?

dhartunian commented on pkg/ui/src/views/cluster/containers/nodesOverview/index.tsx:
```diff
@@ -413,7 +424,7 @@ const decommissionedNodesTableData = createSelector(
   nodesSummarySelector,
   (statuses, nodesSummary): DecommissionedNodeStatusRow[] => {
-    const decommissionedStatuses = statuses.decommissioned || [];
+    const decommissionedStatuses = statuses.live || [];
```
I think this modification is incorrect and it should remain `decommissioned`. Can you take another look?

Maybe add a test to check :)

koorosh replied:
🤦‍♂️oh, that's a terrible result of manual testing where I used "live" nodes to emulate some data in Decommissioned nodes list, and haven't reverted code back.
It is a good point to add tests on these selectors.

dhartunian commented on pkg/ui/src/views/cluster/containers/nodesOverview/index.tsx:
```diff
@@ -133,8 +133,9 @@ class NodeList extends React.Component<LiveNodeListProps> {
-              <Link to={`/node/${record.nodeId}`}>
-                <Text>{record.nodeId}</Text>
+              <Link className="nodes-table__link" to={`/node/${record.nodeId}`}>
+                <Text>{`N`}</Text>
```
Can you take another look at Anne's example? I think the `N` and the number are bolded together.

koorosh replied:
Thank you, fixed.

dhartunian commented on pkg/ui/src/views/cluster/containers/nodesOverview/index.tsx:
```diff
@@ -153,6 +154,7 @@ class NodeList extends React.Component<LiveNodeListProps> {
+      width: "20%",
     },
     {
       key: "nodesCount",
```
Last item on Anne's list was adjustment of this column's naming. I think that's the line below this.

koorosh replied:
You're right. Renamed column name.

dhartunian reviewed (changes requested):
Thanks for the updates @koorosh. A few small adjustments remaining. Let me know if you have any questions.

koorosh committed:
ui: (fix) Styling for Nodes tables on Cluster Overview page

- Make node is with `N` prefix bold together
- Rename `# of nodes` to `node count` column for Live Nodes table

Release note: None

koorosh committed:
ui: Refactoring and tests on state selectors for Nodes Overview page

- `partitionedStatuses` selector has been duplicated in two places
(nodesOverview and nodes reducers modules). The only one is required
(in nodes reducers) because this logic is reused now in several places.
But the last actual logic of this selector has to changed so default
liveness status is "LIVE", because now we display all 'live',
'decommissioning', 'unavailable', and 'dead' nodes in the single table.
- Added test to validate that tables on Node Overview page are supplied
with node records properly partitioned by node status.

Release note: None

dhartunian commented:
bors r+
